felix/labelindex/named_port_index.go

```diff
@@ -462,55 +549,50 @@ func (idx *SelectorAndNamedPortIndex) UpdateIPSet(ipSetID string, sel selector.S
 			}
 			newIPSetData.memberToRefCount[member] = refCount + 1
 		}
-	}
+	})
 }
 
-func (idx *SelectorAndNamedPortIndex) DeleteIPSet(id string) {
-	log.WithField("ipSetID", id).Info("Deleting IP set")
+func (idx *SelectorAndNamedPortIndex) DeleteIPSet(setID string) {
+	ipSetData := idx.ipSetDataByID[setID]
 
-	ipSetData := idx.ipSetDataByID[id]
 	if ipSetData == nil {
-		log.WithField("id", id).Warning("Delete of unknown IP set, ignoring")
+		log.WithField("id", setID).Warning("Delete of unknown IP set, ignoring")
 		return
+	} else {
+		log.WithFields(log.Fields{
+			"ipSetID":  setID,
+			"selector": ipSetData.selector.String(),
+		}).Info("Deleting IP set")
 	}
 
-	// Emit events for all the removed CIDRs.
-	for member := range ipSetData.memberToRefCount {
-		if log.GetLevel() >= log.DebugLevel {
-			log.WithField("member", member).Debug("Emitting deletion event.")
-		}
-		idx.OnMemberRemoved(id, member)
-	}
```

Comment:
Hmm, I thought this had gone in in an earlier tranche of changes.  It was very inefficient to emit all these events when the downstream listener can also see the "IP set deleted" message and clean up en masse.  I'll double check that the corresponding downstream change is in too (if there was one)...

Reply:
Yes, confirmed, the EventSequencer is the thing seeing these events, it buffers them all up in `pendingRemovedIPSetMembers` then it gets the `OnIPSetRemoved` call and it throws that buffer away!

Will also check enterprise handles this, in case there are more callbacks from the named port index.